src/Database/TigerBeetle/Internal/FFI/Account.hsc

```diff
@@ -0,0 +1,93 @@
+{-# LANGUAGE ForeignFunctionInterface #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedRecordDot #-}
+{-# LANGUAGE StrictData #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Database.TigerBeetle.Internal.FFI.Account where
+
+import Data.Word
+import Data.WideWord
+import Foreign.Storable
+
+#include "tb_client.h"
+
+data AccountFlags = 
+      Linked
+    | DebitsMustNotExceedCredits
+    | CreditsMustNotExceedDebits
+    | History
+    | Imported
+    | Closed
+    deriving (Eq, Ord, Show)
+
+instance Enum AccountFlags where
+    fromEnum Linked                     = #const TB_ACCOUNT_LINKED
+    fromEnum DebitsMustNotExceedCredits = #const TB_ACCOUNT_DEBITS_MUST_NOT_EXCEED_CREDITS
+    fromEnum CreditsMustNotExceedDebits = #const TB_ACCOUNT_CREDITS_MUST_NOT_EXCEED_DEBITS
+    fromEnum History                    = #const TB_ACCOUNT_HISTORY
+    fromEnum Imported                   = #const TB_ACCOUNT_IMPORTED
+    fromEnum Closed                     = #const TB_ACCOUNT_CLOSED
+
+    toEnum (#const TB_ACCOUNT_LINKED)                          = Linked
+    toEnum (#const TB_ACCOUNT_DEBITS_MUST_NOT_EXCEED_CREDITS)  = DebitsMustNotExceedCredits 
+    toEnum (#const TB_ACCOUNT_CREDITS_MUST_NOT_EXCEED_DEBITS)  = CreditsMustNotExceedDebits
+    toEnum (#const TB_ACCOUNT_HISTORY)                         = History
+    toEnum (#const TB_ACCOUNT_IMPORTED)                        = Imported
+    toEnum (#const TB_ACCOUNT_CLOSED)                          = Closed
+    toEnum unmatched = error $ "AccountFlags.toEnum: Cannot match " ++ show unmatched
+
+data TBAccount
+  = TBAccount
```

Comment:
```suggestion
data TbAccount
  = TbAccount
```
No need to change the capitalization style, is there?

Reply:
Ahh, this is a good stylistic discussion to get in front of right away. I generally prefer capitalizing abbreviations like TB (Tiger Beetle), or at least keeping the case consistent (i.e. tbAccountId). Thoughts?

Reply:
I'm the opposite but I'm not super strong on it. When I read CamelCase I read `TBAccount` as `TBA` `ccount`. I tend to prefer just following the local style established in the code in the absence of a style guide. That way I can avoid these kinds of changes.

If you feel strongly about it we can make this the official style.

Reply:
Another potential solution is to stop prefixing with TB, since it's redundant. And we can qualify all our imports with FFI. Which I think is a nice indicator anyways!

Reply:
I think we should keep the prefix on the FFI types so that they align with the C types. It's the only module we need to do this for. In general we should design modules to be imported qualified.